### PR TITLE
Create and use script for activating cchq virtualenv

### PR DESCRIPTION
**Pull request by gherceg**
In the case where we want to activate a virtualenv without going through the typical control setup process (aka invoking `init.sh`), we currently attempt to activate `CCHQ_VIRTUALENV` and default to `cchq` if that environment variable is not set. See [here](https://github.com/dimagi/commcare-cloud/blob/master/src/commcare_cloud/commcare_cloud.py#L113-L116) for more context on where this occurs. 

Assuming `CCHQ_VIRTUALENV` variable is not exported in the user's bash profile (which it shouldn't be), that variable is not set until `init.sh` is run. This means we attempt to default to `cchq`. With the shift to `cchq-3.10`, we want this to activate `cchq-3.10` if it exists rather than attempt to activate `cchq`.

The solution is to pull the existing logic from `init.sh` that determines which virtualenv to activate, and create a new script that can be invoked from `init.sh` as well as directly. In the event that the targeted virtualenv does not exist, the script returns with an error. The `--quiet` arg is used from the `init.sh` script since that will handle creating the missing virtualenv, whereas the direct usage expects a failure to output info to the user.

### CODE REVIEW

File: control/activate_venv.sh
Comment (joxl) [inline control/activate_venv.sh:39]: Two things going on here:

1. The previous semicolon position is a syntax error.
2. **Suggestion**: when using an internal "bool" var (whose value you know is either `true` or `false` because your script controls it), it's simpler to just execute the value of the var because the `true` and `false` binaries are always present on all POSIX-compliant OSes.


```suggestion
    if $quiet_mode; then
```

**Review by joxl — CHANGES_REQUESTED** [review]
Change requested for syntax error.

File: control/activate_venv.sh
Comment (joxl) [inline control/activate_venv.sh:28]: nit: prevent head-scratching bugs if whitespace shows up in a path somewhere

```suggestion
        CCHQ_VENV_PATH_OLD="$VENV"
```

File: control/activate_venv.sh
Comment (joxl) [inline control/activate_venv.sh:36]: Same as above (whitespace safety).

```suggestion
if [ -f "$VENV/bin/activate" ]; then
    source "$VENV/bin/activate"
```

File: control/init.sh
Comment (joxl) [inline control/init.sh:29]: Whitespace safety.

```suggestion
    if ! source "${COMMCARE_CLOUD_REPO}/control/activate_venv.sh" --quiet; then
```

File: control/init.sh
Comment (joxl) [inline control/init.sh:39]: Whitespace safety.

```suggestion
                python3.10 -m venv "$VENV"
```

File: control/init.sh
Comment (joxl) [inline control/init.sh:43]: Whitespace safety.

```suggestion
                python3 -m virtualenv "$VENV"
```

File: control/init.sh
Comment (joxl) [inline control/init.sh:46]: Whitespace safety.

```suggestion
        source "$VENV/bin/activate"
```

**Review by joxl — COMMENTED** [review]
A few more suggestions to improve robustness.

File: control/activate_venv.sh
Comment (gherceg) [reply control/activate_venv.sh:39]: > The previous semicolon position is a syntax error.

Distinctly remember writing this and thinking to myself "that's not right" but I clearly did not go back to check 😅 

@skodde and I had [a discussion](https://github.com/dimagi/commcare-cloud/pull/5574#discussion_r1011013702) about bool string values in bash not too long ago. I think you may have differing opinions and my goal is not to ruffle feathers, but to see if we can come to a consensus for consistency's sake. If we chalk it up to personal preference that is also fine!

I do like the idea of `if $quiet_mode; then` in cases where we are absolutely sure the variable will only be "true" or "false", but my question is are we able to guarantee that? Based on my understanding, I'd assume yes but maybe that's wrong.

File: control/init.sh
Comment (gherceg) [reply control/init.sh]: See 577430651879a4e1a537484e6bf7822489082ccf

File: control/init.sh
Comment (gherceg) [reply control/init.sh]: See 577430651879a4e1a537484e6bf7822489082ccf

File: control/init.sh
Comment (gherceg) [reply control/init.sh]: See 577430651879a4e1a537484e6bf7822489082ccf

File: control/init.sh
Comment (gherceg) [reply control/init.sh]: See 577430651879a4e1a537484e6bf7822489082ccf

File: control/activate_venv.sh
Comment (gherceg) [reply control/activate_venv.sh]: Just realized the thread I linked doesn't discuss bools, so I totally made that. It makes sense why we would want to wrap the content of non-bools in quotations since we cannot make assumptions about its value. Sorry for the unnecessary ping!

**Review by joxl — APPROVED** [review]

File: control/activate_venv.sh
Comment (millerdev) [inline control/activate_venv.sh:39]: Should this expression be negated? If not quiet mode, then show a message.

Comment (gherceg) [reply control/activate_venv.sh]: Whoops! See 46c3888819a1d9eaab85f778477a88026ce8ca1b

**Comment by gherceg** [comment]
Needed 4f5f138a9d61781288ad38c432986b1b42dc53b1 after testing this, since `BIONIC_USE_SYSTEM_PYTHON` was unset when calling `activate_venv.sh` directly.

08c119f57709d34552c96fc1c0db38101ad893b6 is mainly cleanup.

File: control/activate_venv.sh
Comment (millerdev) [inline control/activate_venv.sh:38]: I'm a little nervous about this technique since it's dependent on implementation details of the virtualenv provider. For example, I did a similar test in my dev environment:
```sh
$ which python
/home/dimagi/.pyenv/shims/python
$ echo "$VIRTUAL_ENV/bin/python"
/home/dimagi/.pyenv/versions/3.9.12/envs/cchq/bin/python
```
I think typical way to tell if a virtualenv is activated is to test if `$VIRTUAL_ENV` has a value. Alternately, you can just go ahead and `source "$VENV/bin/activate"` unconditionally—activating twice should be fine.

Comment (gherceg) [reply control/activate_venv.sh]: Yeah I misunderstood an issue and chalked it up to this, but you can indeed attempt to activate an active virtualenv without issues. See 343a0d014be3826eb8692b5fc957eb8484cb2fc0

**Comment by gherceg** [comment]
Addressing the test failure in https://github.com/dimagi/commcare-cloud/pull/5626 as it is unrelated to these changes.

**Review by joxl — APPROVED** [review]

File: src/commcare_cloud/commcare_cloud.py
Comment (millerdev) [inline src/commcare_cloud/commcare_cloud.py:124]: Currently `env` will always be non-empty (if `CCHQ_VIRTUALENV` is not set to an empty string) since `default_virtualenv` always has a value. I think some lines above need to change as well:

https://github.com/dimagi/commcare-cloud/blob/7a6622a601c7efc4e3810d73b5b67854384ef34f/src/commcare_cloud/commcare_cloud.py#L115-L116

Line 115 can be removed entirely since we do not support Python 2 anymore. Line 116 should change to
```py
    venv = os.environ.get("CCHQ_VIRTUALENV")
```

**Review by millerdev — CHANGES_REQUESTED** [review]

File: src/commcare_cloud/commcare_cloud.py
Comment (millerdev) [reply src/commcare_cloud/commcare_cloud.py:123]: Update: I forgot that the activation script appends `-3.10` to the value of `CCHQ_VIRTUALENV` unless that suffix is already there (or unless `BIONIC_USE_SYSTEM_PYTHON=true`), so my comment was wrong and therefore not a blocker. However, the cleanup would still be nice.

**Review by millerdev — APPROVED** [review]

File: src/commcare_cloud/commcare_cloud.py
Comment (gherceg) [reply src/commcare_cloud/commcare_cloud.py:123]: Thanks. See 9e0ec6c10fffc0507a69a8eb4c5a5ce8d89a0296.

Offlined with Daniel and clarified that we still want to allow callers to pass a value in for `CCHQ_VIRTUALENV` if it is set, but agreed there is no need to provide a default since the `activate_venv.sh` script handles that, and the `ansible` venv has long been irrelevant.

**Comment by gherceg** [comment]
My last commit just added a space in the command and does not impact any behavior, so I'm not going to wait for approval on that.

**Review by millerdev — APPROVED** [review]

File: src/commcare_cloud/commcare_cloud.py
Comment (joxl) [inline src/commcare_cloud/commcare_cloud.py:123]: I think this causes a "discontinuity problem" when executing `cchq --control ...` _from_ a system with this change (this PR) _against_ a control environment that does not have this change (yet).  In that scenario, commands are executed against the remote control system that depend on this file (`activate_venv.sh`) being present, but since it isn't, the command crashes.

Comment (gherceg) [reply src/commcare_cloud/commcare_cloud.py:123]: Thanks for flagging this Joel. My initial thought is that a change like this should have had a changelog/commcare-cloud announcement with it as well. I'll put that together now.

Comment (gherceg) [reply src/commcare_cloud/commcare_cloud.py:123]: See https://github.com/dimagi/commcare-cloud/pull/5649

**Pull request merged**
